src/btree/bt_handle.c

```diff
@@ -551,12 +554,14 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
      *
      * Rollback to stable does not operate on logged tables and metadata, so it is skipped.
      *
-     * The only scenario where the checkpoint run write generation number is less than the
-     * connection last checkpoint base write generation number is when rollback to stable doesn't
-     * happen during the recovery due to the unavailability of history store file.
+     * The only scenarios where the checkpoint run write generation number is less than the
+     * connection last checkpoint base write generation number are when rollback to stable doesn't
+     * happen during the recovery due to the unavailability of history store file, or when reading a
+     * checkpoint.
      */
-    if (!F_ISSET(conn, WT_CONN_RECOVERING) || F_ISSET(btree, WT_BTREE_LOGGED) ||
-      ckpt->run_write_gen < conn->last_ckpt_base_write_gen)
+    if ((!F_ISSET(conn, WT_CONN_RECOVERING) || F_ISSET(btree, WT_BTREE_LOGGED) ||
+          ckpt->run_write_gen < conn->last_ckpt_base_write_gen) &&
+      !is_ckpt)
```

Comment:
We shouldn't avoid setting the `base_write_gen` for unnamed checkpoints also. As it is possible that unnamed checkpoints don't write new checkpoints in the new session if there are no changes. So don't set it only for the named checkpoints.

Reply:
(This comment is out of date, right?)

Reply:
Yes, Thanks.